Actions/AL-Go-Helper.ps1

```diff
@@ -2360,4 +2360,81 @@ function Get-PackageVersion($PackageName) {
     else {
         throw "Package $PackageName is not in the list of packages"
     }
-}
\ No newline at end of file
+}
+
+function InstallAzModuleIfNeeded {
+    Param(
+        [string] $moduleName
+    )
+
+    if (Get-Module -Name $moduleName) {
+        # Already installed
+        return
+    }
+    if ($isWindows) {
+        # GitHub hosted Windows Runners have AZ PowerShell module saved in C:\Modules\az_*
+        # Remove AzureRm modules from PSModulePath and add AZ modules
+        if (Test-Path 'C:\Modules\az_*') {
+            $azModulesPath = Get-ChildItem 'C:\Modules\az_*' | Where-Object { $_.PSIsContainer }
+            if ($azModulesPath) {
+              Write-Host "Adding AZ module path: $($azModulesPath.FullName)"
+              $ENV:PSModulePath = "$($azModulesPath.FullName);$(("$ENV:PSModulePath".Split(';') | Where-Object { $_ -notlike 'C:\\Modules\Azure*' }) -join ';')"
+            }
+        }
+    }
+    else {
+        # Linux runners have AZ PowerShell module saved in /usr/share/powershell/Modules/Az.*
+    }
+    $azModule = Get-Module -name $moduleName -ListAvailable | Select-Object -First 1
+    if ($azModule) {
+        Write-Host "$moduleName Module is available in version $($azModule.Version)"
```

Comment:
Shouldn't we at least have some MinimumVersion parameter we require? Since we support self-hosted runner we have no idea which versions of Az modules those runners have. There's a risk that those runners have an older version of Az that doesn't support what we need.

Reply:
We are using the bare minimum of Az Storage (we actually also work with AzureRm-Storage module (which is what came before az.storage) - so we do work with all versions of Az.Storage.

Reply:
We're also using Az.Accounts and Az.KeyVault. Does AL-Go work with **all** older versions of these modules as well?

And even if we do, I still think it'd be a good idea to require a minimum version. If we ever change how we invoke e.g. Az.Accounts I don't want to have to verify that it works with **all** versions of the module. It'd be easier for us if we can just verify that it works with some minimum version of Az.Accounts and then hope that the Az team doesn't break us in any future versions 😄

Reply:
I have added all referenced PowerShell modules to Packages.json and using the version specified there as a minimum version.
This also means that Packages.json needs to be downloaded in localdevenv and clouddevenv (as they might install az.KeyVault) - this is added as well.
Currently all references is 1.0.0 - if we ever need a newer version, we should update it there.